pipeline/utils/github_utils.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utility functions related to GitHub interaction. Uses the GitHub API with
+reference https://developer.github.com/v3/"""
+
+import requests
+import json
+import os
+
+_API_URL = 'https://api.github.com'
+# file modes for different git file types
+_TREE_EL_MODES = {'file': u'100644',
+                  'executable': u'100755',
+                  'tree': u'040000',
+                  'symlink': u'120000'}
+_REQUEST_FUNCS = {'GET': requests.get,
+                  'POST': requests.post,
+                  'PATCH': requests.patch}
+
+
+def _is_executable(el):
+    return os.path.isfile(el) and os.access(el, os.X_OK)
+
+
+def _get_immediate_subdirs(a_dir):
+    return filter(lambda x: os.path.isdir(os.path.join(a_dir, x)),
+                  os.listdir(a_dir))
+
+
+def _get_immediate_files(a_dir):
+    return filter(lambda x: os.path.isfile(os.path.join(a_dir, x)),
+                  os.listdir(a_dir))
+
+
+def _exec_request(http_method, url, req_data=None, username=None,
+                  password=None):
+    """Make an HTTP request with given data and (username, password) tuple.
+    Return json response if request succeeds, otherwise raise an exception"""
+    request_func = _REQUEST_FUNCS[http_method]
```

Comment:
See comment above. Would prefer `http_method` were just in `[requests.get, requests.post, requests.patch]` so that you don't need to translate string params